actions/helm_stage_e2e.go

```diff
@@ -0,0 +1,129 @@
+package actions
+
+import (
+	"fmt"
+	"io/ioutil"
+	"log"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/codegangsta/cli"
+	"github.com/google/go-github/github"
+)
+
+const (
+	// RepoFlag represents the '-repo' flag
+	RepoFlag = "repo"
+	// RefFlag represents the '-ref' flag (for specifying a SHA, branch or tag)
+	RefFlag = "ref"
+	// GHOrgFlag represents the '-ghOrg' flag
+	GHOrgFlag = "ghOrg"
+)
+
+// HelmStageE2E is the cli handler for generating a release helm chart for deis-e2e
+func HelmStageE2E(ghClient *github.Client) func(*cli.Context) {
+	return func(c *cli.Context) {
+		const chartDir = "workflow-dev-e2e"
+		downloadFiles(ghClient, c, chartDir, []string{"README.md", "Chart.yaml"})
+
+		// set up staging dir
+		createDir(stagingDirName)
+
+		if deisRelease.Full != "" && deisRelease.Short != "" {
+			if err := updateFilesWithRelease(chartDir); err != nil {
+				log.Fatalf("Error updating files with release (%s)", err)
+			}
+		} else {
+			log.Println("DEIS_RELEASE and/or DEIS_RELEASE_SHORT not provided so not amending staged files.")
+		}
+	}
+}
+
+func createDir(dirName string) {
+	if _, err := os.Stat(dirName); os.IsNotExist(err) {
+		if err := os.Mkdir(dirName, os.ModePerm); err != nil {
+			log.Fatalf("Error creating dir %s (%s)", dirName, err)
+		}
+	}
+}
+
+func getStagingPath(dirName string) string {
+	return fmt.Sprintf("%s/%s", stagingDirName, dirName)
+}
+
+func downloadFiles(ghClient *github.Client, c *cli.Context, rootDirName string, fileNames []string) {
+	var opt github.RepositoryContentGetOptions
+	opt.Ref = c.GlobalString(RefFlag)
+
+	createDir(fmt.Sprintf("%s/%s", stagingDirName, rootDirName))
+
+	for _, fileName := range fileNames {
+		filePath := fmt.Sprintf("%s/%s", rootDirName, fileName)
+		readCloser, err := downloadContents(ghClient, c.GlobalString(GHOrgFlag), c.GlobalString(RepoFlag), filePath, &opt)
+		if err != nil {
+			log.Fatalf("Error downloading contents of %s (%s)", filePath, err)
+		}
+		defer readCloser.Close()
+		contents, err := ioutil.ReadAll(readCloser)
+		if err != nil {
+			log.Fatalf("Error reading contents of %s (%s)", filePath, err)
+		}
+		// create file in stagingDirName and write contents to it
+		currentWorkingDir, err := os.Getwd()
+		if err != nil {
+			log.Fatalf("Error getting current working dir (%s)", err)
+		}
+		newFileName := fmt.Sprintf("%s/%s/%s", currentWorkingDir, stagingDirName, filePath)
+		f, err := os.Create(newFileName)
+		if err != nil {
+			log.Fatalf("Error creating file %s (%s)", newFileName, err)
+		}
+		defer f.Close()
+		if _, err := f.Write(contents); err != nil {
+			log.Fatalf("Error writing contents to file %s (%s)", newFileName, err)
+		}
```

Comment:
note that you can replace the `ioutil.ReadAll` and `f.Write` with an `io.Copy(f, readCloser)`. mind doing that here?